### PR TITLE
Add iconColor to ha-config-navigation entries

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Screenshot from 2021-11-19 11-18-23

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/dashboard/ha-config-navigation.ts [hunk 1]
```diff
@@ -24,10 +24,13 @@ class HaConfigNavigation extends LitElement {
           ? html`
               <a href=${page.path} aria-role="option" tabindex="-1">
                 <paper-icon-item>
-                  <ha-svg-icon
-                    .path=${page.iconPath}
+                  <div
```
Comment (balloob) [inline src/panels/config/dashboard/ha-config-navigation.ts:27]: The circle containers are not completely square

image

File: gallery/src/demos/demo-ha-config-navigation.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    path: "#demo-ha-config-navigation",
+    name: "Basic icon with no background",
+    description:
+      "This is just here to display that the component can handle this to",
+    iconPath: mdiRocket,
+  },
+];
+
+@customElement("demo-ha-config-navigation")
+export class DemoHaConfigNavigation extends LitElement {
```
Comment (balloob) [inline gallery/src/demos/demo-ha-config-navigation.ts:42]: I'm a bit 🤷 on this gallery page. Like… it's used on one page and requires no set-up to see how it renders things. It's not a component that we reuse.

**Comment by spacegaier** [comment]
I think we need to make those icon colors themeable, because the dark blue and dark green ones (basically the first 10 on the config page from the screenshot), will be trouble in the dark theme mode I think.

**Review by balloob — APPROVED** [review]
Looks good! Ok to merge when dark mode colors added.

@matthiasdebaat can you provide some color codes for dark mode to use?

**Comment by ludeeus** [comment]
I don't think the BG colors will have an issue.
But the icon inside should be white(-ish) color (at least by default) for both light and dark
image

**Comment by balloob** [comment]
Ah yes, that looks indeed fine. SHIP IT. We can tweak it later.

**Comment by matthiasdebaat** [comment]
Will create a better color pallet, including dark mode colors. I picked the current ones just randomly with a color picker.

**Pull request merged**